### DISCUSSION TITLE
Question: are the parsers audited?

### DISCUSSION THREAD

**Discussion by yuriescl** [nvim-treesitter/nvim-treesitter#6614] [General]
I see most parsers are C code and are maintained by many different maintainers. I wonder if there is any kind of audit of what's inside the parser C code. Since C code can be anything it can be malicious as well.

**Comment by lucario387** [comment]
The C code is completely optional :) You can always generate the C code yourself. The source is the `grammar.js` file

**Comment by yuriescl** [comment]
Is there an easy way to point the parser of a specific language to be a custom GitHub repo (a fork, for example)? Or do I have to fork this repo and modify parsers.lua directly?

**Reply by clason** [reply]
This is documented, please read the README (advanced setup).

And, no, there is no auditing; parsers are provided as-is without warranty; *caveat emptor*. This plugin should _not_ be used by people with low risk tolerance (or heightened security requirements). Things may -- or may not -- get better when WASM parsers are supported (in half a year or so).

**Comment by lucario387** [comment]
From nvim-treesitter we have `TSInstallFromGrammar` (requires `tree-sitter` executable) that will do that part for you :)

For pointing to a specific commit SHA you can always do it yourself as per README.md
